htdocs/core/lib/functions.lib.php

```diff
@@ -6085,7 +6085,7 @@ function getCommonSubstitutionArray($outputlangs, $onlykey=0, $exclude=null, $ob
 					if (is_array($extrafields->attributes[$object->table_element]['label']) && count($extrafields->attributes[$object->table_element]['label']) > 0)
 					{
 						foreach ($extrafields->attributes[$object->table_element]['label'] as $key => $label) {
-							$substitutionarray['__EXTRAFIELD_' . strtoupper($key) . '__'] = $object->array_options['options_' . $key];
+							$substitutionarray['__EXTRAFIELD_' . strtoupper($key) . '__'] = $extrafields->showOutputField($key, $object->array_options['options_' . $key]);
```

Comment:
The method showOutputField is a method that format a data for html output.
The substitution array must contains the real brute value as it can be used for any type of used (formatting must be done during the output if we can and avoid in memory that stores data). When we need a formatting (for example to get format according to language, the substitution array has a suffix into name, for example "_locale".

Why do you need to format the data inside the array in memory ?